src/cmake/messages.cmake

```diff
@@ -95,3 +95,65 @@ function(messages)
     WORKING_DIRECTORY ${SOURCE_DIR}
   )
 endfunction()
+
+function(messages_multi)
```

Comment:
This appears to copy the `messages` function just to add a new DEPENDS argument.  Could you just make take an option on the current one and handle if it is empty?